internal/service/loadbalancer/loadbalancer.go

```diff
@@ -2,544 +2,660 @@ package loadbalancer
-		Plan:             d.Get("plan").(string),
-		Zone:             d.Get("zone").(string),
-		NetworkUUID:      d.Get("network").(string),
-		ConfiguredStatus: upcloud.LoadBalancerConfiguredStatus(d.Get("configured_status").(string)),
-		Frontends:        []request.LoadBalancerFrontend{},
-		Backends:         []request.LoadBalancerBackend{},
-		Resolvers:        []request.LoadBalancerResolver{},
-		Networks:         networks,
-		Labels:           utils.LabelsMapToSlice(d.Get("labels").(map[string]interface{})),
-		MaintenanceDOW:   upcloud.LoadBalancerMaintenanceDOW(d.Get("maintenance_dow").(string)),
-		MaintenanceTime:  d.Get("maintenance_time").(string),
-	}
-	lb, err := svc.CreateLoadBalancer(ctx, req)
-	if err != nil {
-		return utils.HandleResourceError(d.Get("name").(string), d, err)
-	}
-
-	d.SetId(lb.UUID)
+func (r *loadBalancerResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data loadBalancerModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
 
-	if diags = setLoadBalancerResourceData(d, lb); len(diags) > 0 {
-		return diags
+	if resp.Diagnostics.HasError() {
+		return
 	}
-
-	tflog.Info(ctx, "load balancer created", map[string]interface{}{"name": lb.Name, "uuid": lb.UUID})
-	return diags
-}
-
-func resourceLoadBalancerRead(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
-	var err error
-	svc := meta.(*service.Service)
-	lb, err := svc.GetLoadBalancer(ctx, &request.GetLoadBalancerRequest{UUID: d.Id()})
-	if err != nil {
-		return utils.HandleResourceError(d.Get("name").(string), d, err)
+	var labelsMap map[string]string
+	if !data.Labels.IsNull() && !data.Labels.IsUnknown() {
+		resp.Diagnostics.Append(data.Labels.ElementsAs(ctx, &labelsMap, false)...)
 	}
+	labels := utils.NilAsEmptyList(utils.LabelsMapToSlice(labelsMap))
 
-	if diags = setLoadBalancerResourceData(d, lb); len(diags) > 0 {
-		return diags
-	}
+	var networks []request.LoadBalancerNetwork
+	if !data.Networks.IsNull() && !data.Networks.IsUnknown() {
+		var networkModels []loadbalancerNetworkModel
+		resp.Diagnostics.Append(data.Networks.ElementsAs(ctx, &networkModels, false)...)
 
-	return diags
-}
-
-func resourceLoadBalancerUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
-	svc := meta.(*service.Service)
-
-	// handle network renaming before modifying load balancer so that new network names are present in `lb` before setting state
-	if diags = resourceLoadBalancerNetworkUpdate(ctx, d, svc); len(diags) > 0 {
-		return diags
+		for i, n := range networkModels {
+			network := request.LoadBalancerNetwork{
+				Name:   n.Name.ValueString(),
+				Type:   upcloud.LoadBalancerNetworkType(n.Type.ValueString()),
+				Family: upcloud.LoadBalancerAddressFamily(n.Family.ValueString()),
+				UUID:   n.Network.ValueString(),
+			}
+			if network.Type == upcloud.LoadBalancerNetworkTypePrivate && network.UUID == "" {
```

Comment:
`AlsoRequires` validator might work for this as well 🤔 Validator would anyway be nicer way to check this, but I guess it would require custom validator so maybe better to look at that later

Reply:
Added a plan modifier, so we can check & validate the config after the possible defaults have been set.